pkg/user/tx_client.go

```diff
@@ -157,9 +158,16 @@ type TxClient struct {
 	// that was submitted to the chain
 	txTracker           map[string]txInfo
 	gasEstimationClient gasestimation.GasEstimatorClient
+	// trustedMode indicates whether to trust the consensus node for account information.
+	// When true (default), the client trusts the consensus node to provide accurate
+	// account numbers and sequences without additional verification.
+	// When false, additional verification steps would be performed (to be implemented).
```

Comment:
Do we need a follow-up issue to track implementing the additional verification steps?

```suggestion
	// When false, additional verification steps need to be performed (to be implemented).
```